eng/common/scripts/get-codeowners.ps1

```diff
@@ -1,49 +1,71 @@
-  $RootDirectory, # ideally $(Build.SourcesDirectory)
-  $VsoVariable = "" # target devops output variable
+  [string]$TargetDirectory, # Code path to code owners. e.g sdk/core/azure-amqp
+  [string]$RootDirectory = "$env:SYSTEM_DEFAULTWORKINGDIRECTORY", # The repo contains CODEOWNER file.
+  [string]$CodeOwnerFileLocation = "$PSSCriptRoot/../../../.github/CODEOWNERS", # The absolute path of CODEOWNERS file. 
+  [string]$ToolVersion = "1.0.0-dev.20211118.20", # Placeholder. Will update in next PR
+  [string]$ToolPath = "$env:AGENT_TOOLSDIRECTORY", # The place to check the tool existence. Put $(Agent.ToolsDirectory) as default
+  [string]$DevOpsFeed = "https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v3/index.json", # DevOp tool feeds.
+  [string]$VsoVariable = "" # Option of write code owners into devop variable
 )
-$target = $TargetDirectory.ToLower().Trim("/")
-$codeOwnersLocation = Join-Path $RootDirectory -ChildPath ".github/CODEOWNERS"
-$ownedFolders = @{}
 
-if (!(Test-Path $codeOwnersLocation)) {
-  Write-Host "Unable to find CODEOWNERS file in target directory $RootDirectory"
-  exit 1
+$LocalCodeOwnerPath = ".github/CODEOWNERS"
+$ToolCommandName = "retrieve-codeowners"
+$ToolName = "Azure.Sdk.Tools.RetrieveCodeOwners"
+function New-TemporaryDirectory {
+  $parent = [System.IO.Path]::GetTempPath()
+  [string] $name = [System.Guid]::NewGuid()
+  [string] $newPath = Join-Path $parent $name
+  New-Item -ItemType Directory -Path $newPath | Out-Null
+  return $newPath
 }
 
-$codeOwnersContent = Get-Content $codeOwnersLocation
-
-foreach ($contentLine in $codeOwnersContent) {
-  if (-not $contentLine.StartsWith("#") -and $contentLine){
-    $splitLine = $contentLine -split "\s+"
-    
-    # CODEOWNERS file can also have labels present after the owner aliases
-    # gh aliases start with @ in codeowners. don't pass on to API calls
-    $ownedFolders[$splitLine[0].ToLower().Trim("/")] = ($splitLine[1..$($splitLine.Length)] `
-      | ? { $_.StartsWith("@") } `
-      | % { return $_.substring(1) }) -join ","
+function InstallRetrieveCodeOwnersTool() {
+  # Check if the retrieve-codeowners tool exsits or not.
+  if (Get-Command "$ToolPath/$ToolCommandName" -errorAction SilentlyContinue) {
+    return "$ToolPath/$ToolCommandName"
   }
-}
+  if (!$ToolPath -or !(Test-Path $ToolPath)) {
+    $ToolPath = New-TemporaryDirectory
+  }
+  Write-Warning "Installing the ToolCommandName tool under $ToolPath... "
+  dotnet tool install --tool-path $ToolPath --add-source $DevOpsFeed --version $ToolVersion $ToolName | Out-Null
 
-$results = $ownedFolders[$target]
+  return Join-Path $ToolPath $ToolCommandName
+}
 
-if ($results) {
-  Write-Host "Found a folder $results to match $target"
+function GetCodeOwners ([string] $Command) {
+  # Run code owner tools to retrieve code owners.
+  if (!(Get-Command $Command -errorAction SilentlyContinue)) {
```

Comment:
This check should be added in the Get-CodeOwnersTool function and so you don't need to call it from both.